.github/workflows/integration-tests.yml

```diff
@@ -40,10 +40,8 @@ concurrency:
   # The "no-run" group ensures that irrelevant label events don't interfere with the real workflows.
   group: >-
     ${{ github.workflow }}-${{ github.ref }}-${{
-    (github.event.action == 'opened' ||
-     github.event.action == 'synchronize' ||
-     (github.event.action == 'labeled' && contains(github.event.pull_request.labels.*.name, 're-record-tests'))) &&
-    (contains(github.event.pull_request.labels.*.name, 're-record-tests') && 'rerecord' || 'replay') ||
+    (github.event.action == 'opened' || github.event.action == 'synchronize') && 'replay' ||
+    (github.event.action == 'labeled' && contains(github.event.pull_request.labels.*.name, 're-record-tests')) && 'rerecord' ||
```

Comment:
Your change looks reasonable to me. I wish I could write this craziness more simply. However, for now, I am going to edit your PR just a bit and add a couple of parentheses to help us all out.